src/components/Form/Form.js

```diff
@@ -1,19 +1,27 @@
 import {useRouter} from 'next/router';
+import {useEffect, useState} from 'react';
 
 import useStore from '../../hooks/useStore';
+import Button from '../Button/Button';
 
 import StyledForm from './StyledForm';
 
-export default function Form() {
+export default function Form({task = {}}) {
+	const [taskName, setTaskName] = useState(task.name ?? '');
+	const [duration, setDuration] = useState(task.time ?? '');
 	const addTask = useStore(state => state.addTask);
+	const editTask = useStore(state => state.editTask);
 	const router = useRouter();
 
+	useEffect(() => {
+		setTaskName(task.name ?? '');
+		setDuration(task.time ?? '');
+	}, [task]);
+
 	function handleSubmit(event) {
 		event.preventDefault();
 		const form = event.target;
-		const taskName = form.elements.task.value.trim();
-		const duration = Number.parseInt(form.elements.duration.value, 10);
-		addTask(taskName, duration);
+		task.id ? editTask(task.id, taskName, duration) : addTask(taskName, duration);
```

Comment:
You would use the ternary operator only if value es generated - which is not the case here (like: `const myVar = ...? ... : ...`)

Please use an if-else-block